### PR TITLE
Finished merge and added 'My Reports' for agents

### PR DESCRIPTION
**Pull request by sandriskupavskisamex**

### CODE REVIEW

File: AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/PermissionsVerifier.java [hunk 1]
```diff
@@ -30,6 +30,9 @@ public void verifyIsAdminOrRelevantUser(Request req, int relevantUserId) {
     public void verifyIsAdminOrRelevantAgent(Request req, int relevantAgentId) {
         verifyUser(req, user -> user.isAdmin() || (user.getAgentId() != null && user.getAgentId() == relevantAgentId));
     }
+    public void verifyIsAdminOrAgent(Request req) {
```
Comment (JackMead) [inline AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/core/PermissionsVerifier.java]: Can we get a line break above this? :)

File: AgentDiscoveries-Backend/src/main/resources/config.properties [hunk 2]
```diff
@@ -1,20 +0,0 @@
-
```
Comment (JackMead) [inline AgentDiscoveries-Backend/src/main/resources/config.properties]: Guessing you didn't mean to check this in - looks like something weird had happened with git when we were looking at this file. Can we undo this?

File: AgentDiscoveries-Frontend/app/src/components/nav.jsx [hunk 3]
```diff
@@ -105,14 +105,24 @@ export default class NavigationBar extends React.Component {
...
+            <React.Fragment>
+                <NavDropdown eventKey={4} title='Submit' id='basic-nav-dropdown'>
+                    <MenuItem componentClass={Link} href='/submit/location' to='/submit/location' eventKey={4.1}>
+                        Location Report
+                    </MenuItem>
+                    <MenuItem componentClass={Link} href='/submit/region' to='/submit/region' eventKey={4.2}>
+                        Region Summary
+                    </MenuItem>
+                </NavDropdown>
+                <NavDropdown eventKey={5} title='My Reports' id='basic-nav-dropdown'>
```
Comment (JackMead) [inline AgentDiscoveries-Frontend/app/src/components/nav.jsx]: Annoyingly, there's already an event key 5 being used I think! They seem to be all over the place :(

File: AgentDiscoveries-Frontend/app/src/components/search-forms/agent-region-summaries.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import QueryString from 'query-string';
+import {apiGet} from '../utilities/request-helper';
+
+export default class AgentRegionSummaries extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            results: [{}],
+            message: {}
```
Comment (JackMead) [inline AgentDiscoveries-Frontend/app/src/components/search-forms/agent-region-summaries.jsx]: It looks like we never use this message?

File: AgentDiscoveries-Frontend/app/src/components/search-forms/agent-location-reports.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import QueryString from 'query-string';
+import {apiGet} from '../utilities/request-helper';
+
+export default class AgentLocationReports extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            results: [{}],
+            message: {}
```
Comment (JackMead) [inline AgentDiscoveries-Frontend/app/src/components/search-forms/agent-location-reports.jsx]: Again, looks like we never use this message

File: AgentDiscoveries-Backend/src/main/resources/config.properties (hunk 2 above)
Comment (sandriskupavskisamex) [reply AgentDiscoveries-Backend/src/main/resources/config.properties]: I am not sure how to undo this. (confused face)

**Pull request merged**